### PR TITLE
Make scripts compatible with macOS/BSD

### LINKED ISSUE #112: Unable to build on macOS
I am unable to build the webvault on macOS with node+npm. 
To reproduce, follow the build instruction on macOS.

```
bw_web_builds % make full    
./scripts/checkout_web_vault.sh
Input a git ref (commit hash, branch name, tag name, 'master'): master
Using: 'master' to checkout bitwarden/client.
~/bw_web_builds/web-vault ~/bw_web_builds
Already on 'master'
Your branch is up to date with 'origin/master'.
HEAD is now at 4e4de9812 Bumped web version to 2023.2.0 (#4753)
Your branch is up to date with 'origin/master'.
~/bw_web_builds
~/bw_web_builds/web-vault ~/bw_web_builds
From https://github.com/bitwarden/clients
 * branch                master     -> FETCH_HEAD
Already up to date.
Already on 'master'
Your branch is up to date with 'origin/master'.
~/bw_web_builds
./scripts/patch_web_vault.sh
~/bw_web_builds/web-vault ~/bw_web_builds
Patch file not found, using latest
find: -printf: unknown primary or operator
FAILED: line 15, exit code 1. Press any key to exit...
make: *** [patch-web-vault] Error 1
```

This seems to be caused by the missing `-printf` on `find` provided by macOS(or FreeBSD, since macOS the `find` command is based on FreeBSD one).

### PR DESCRIPTION
**Pull request by BlackDex**
All the scripts used by this repo were more focused on Linux based systems. They did not work on macOS or other BSD based systems because of flags or commands which are not available.

This PR resolves those items by adjusting the failed commands so they work on both Linux and macOS/BSD systems.

Fixes #112

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -53,7 +53,7 @@ WORKDIR /vault/apps/web
 RUN npm run dist:oss:selfhost
 
 RUN printf '{"version":"%s"}' \
-      $(git -c 'versionsort.suffix=-' ls-remote --tags --sort='v:refname' https://github.com/dani-garcia/bw_web_builds.git 'v*' | tail -n1 | sed -E 's#.*?refs/tags/v##') \
+      $(git -c 'versionsort.suffix=-' ls-remote --tags --sort='v:refname' https://github.com/dani-garcia/bw_web_builds.git 'v*' | tail -n1 | grep -Eo '[^\/]*$') \
```
Comment (tessus) [inline Dockerfile:56]: Unfortunately this introduced issue #115

File: scripts/build_web_vault.sh [hunk 2]
```diff
@@ -30,7 +30,7 @@ npm run dist:oss:selfhost
 
 # Create vw-version.json with the latest tag from the remote repo.
 printf '{"version":"%s"}' \
-      "$(git -c 'versionsort.suffix=-' ls-remote --tags --sort='v:refname' https://github.com/dani-garcia/bw_web_builds.git 'v*' | tail -n1 | sed -E 's#.*?refs/tags/v##')" \
+      "$(git -c 'versionsort.suffix=-' ls-remote --tags --sort='v:refname' https://github.com/dani-garcia/bw_web_builds.git 'v*' | tail -n1 | grep -Eo '[^\/]*$')" \
```
Comment (tessus) [inline scripts/build_web_vault.sh:33]: Unfortunately this introduced issue https://github.com/dani-garcia/bw_web_builds/issues/115

Comment (BlackDex) [reply scripts/build_web_vault.sh:33]: Ah, the v is still there. While not a very big issue that should be fixed indeed.

Comment (tessus) [reply scripts/build_web_vault.sh:33]: yep, see #116

**Pull request merged**